java/src/main/java/com/nedap/retail/example/Application.java

```diff
@@ -167,41 +173,67 @@ private static void sendHeartbeat(final BufferedReader inputBuffer) throws Excep
 
     private static void retrieveSystemInfo() throws Exception {
         final SystemInfo info = api.retrieveSystemInfo();
-        LOG.info("System info");
-        LOG.info("System id: {}", info.id);
-        LOG.info("Firmware version: {}", info.version);
-        LOG.info("System role: {}", info.systemRole);
-        LOG.info("System time: {}", info.systemTime);
+        if (info != null) {
+            LOG.info("System info");
+            LOG.info("System id: {}", info.id);
+            LOG.info("Firmware version: {}", info.version);
+            LOG.info("System role: {}", info.systemRole);
+            LOG.info("System time: {}", info.systemTime);
+        } else {
+            throw new MessageParsingException("System info could not be retrieved");
+        }
+    }
+
+    private static void retrieveGroupInfo(final GroupInfo.Group group) {
```

Comment:
This method does not retrieve anything. I suggest to rename it to `logGroupInfo`